### PR TITLE
Format FILTER expressions

### PR DESCRIPTION
**Pull request by scimetfoo**
Addresses #9

### CODE REVIEW

File: src/honeysql_postgres/format.cljc [hunk 1]
```diff
@@ -113,6 +114,13 @@
   (str "DO UPDATE SET " (sqlf/comma-join (for [[k v] values]
                                            (str (sqlf/to-sql k) " = " (sqlf/to-sql v))))))
 
+(defmethod format-clause :filter [[_ expr] sql-map]
```
Comment (kitallis) [inline src/honeysql_postgres/format.cljc:119]: What do you think of something like this?

```clojure
(defmethod format-clause :filter [[_ expr] sql-map]
  (let [format-fn (fn [expr]
                    (let [[expression clause alias] (mapv sqlf/to-sql expr)]
                      (->> (expression)
                           (str " FILTER ")
                           (clause)
                           (as-> alias
                                 (str " AS " alias)))))]
    (->> expr
         (map format-fn)
         (sqlf/comma-join)
         (as-> (seq (:select sql-map))
               (str ", ")))))

```

I don't mind using `as->` for simple string munging nil-safe operations.

File: test/honeysql_postgres/postgres_test.cljc [hunk 2]
```diff
@@ -65,6 +96,17 @@
                          :do-update-set {:fields [:phone :name]
                                          :where  [:= :user.active false]}}}))))
 
+(deftest filter-test
+  (is (= ["count(*) FILTER (WHERE NOT i BETWEEN ? AND ?) AS a" 3 5]
+         (sql/format (filter [(sql/call :count :*)(where [:not [:between :i 3 5]]) :a]))))
+
+  (is (= ["SELECT count(*) , count(*) FILTER (WHERE s.i < ?) AS foo, count(*) FILTER (WHERE s.i BETWEEN ? AND ?) AS bar FROM generate_series(1,10) AS s(i)" 5 3 10]
```
Comment (kitallis) [inline test/honeysql_postgres/postgres_test.cljc:73]: There's an additional space after the first `SELECT count(*) ,` clause. 

Is that because of us [back-placing](https://github.com/nilenso/honeysql-postgres/pull/42/files#diff-f81b5b6904b08f9c50a240ecab26f660e4f13c113064b4519324a08f69ac62f8R121) the `,` if it's a select clause or has it always existed before?

File: deps.edn [hunk 3]
```diff
@@ -1 +1 @@
-{:deps {honeysql {:mvn/version "0.9.5" :exclusions [org.clojure/clojurescript]}}}
+{:deps {honeysql {:mvn/version "1.0.444" :exclusions [org.clojure/clojurescript]}}}
```
Comment (kitallis) [inline deps.edn]: Should we do this upgrade in isolation?

File: README.md [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
+
+The query creation and usage is exactly the same as honeysql.
 
 ## Index
 
 - [Usage](#usage)
   - [Leiningen](#leiningen)
   - [Maven](#maven)
   - [REPL](#REPL)
-  - [Breaking Change](#breaking-change)
```
Comment (kitallis) [inline README.md:13]: 👍🏼

**Review by kitallis — COMMENTED** [review]
Looks mostly good, put some minor thoughts in.

File: src/honeysql_postgres/format.cljc (hunk 1 above)
Comment (scimetfoo) [reply src/honeysql_postgres/format.cljc:119]: Couple of things I want to note:

1. Nested threaded macros are generally hard to follow, and especially so when the arguments are flipping positions. I try to avoid macros entirely within a threaded expression. 
2. The `as->` expression here will have to be bound to something (I think that’s been missed), and that binding will go unused. In the first `as->` expression, the alias doesn’t undergo transformations. And in the second `as->` expression, the binding would be redundant since it won't be used.
3. With string concatenation/formatting that requires certain strings to be concatenated/formatted conditionally, it’s generally easier to follow when the arguments are in-place, as opposed to when it’s done using threaded macros. 

Let me know if I’ve missed the rationale of doing it this way or if you have other thoughts about the things I’ve noted here.

File: test/honeysql_postgres/postgres_test.cljc (hunk 2 above)
Comment (scimetfoo) [reply test/honeysql_postgres/postgres_test.cljc:73]: Looks like it gets added in before this expression is formatted. There's a space that gets added in between each different clause that's formatted, which is the cause of this. I see this happening with a couple more clauses too. I'll dig deeper and open up an issue.

File: deps.edn (hunk 3 above)
Comment (scimetfoo) [reply deps.edn]: I'd bumped it up to work with the latest clause priorities.

File: test/honeysql_postgres/postgres_test.cljc (hunk 2 above)
Comment (kitallis) [reply test/honeysql_postgres/postgres_test.cljc:73]: Cool, let's open a separate issue and keep a track. Doesn't need to go in here since it didn't originate here.

File: src/honeysql_postgres/format.cljc (hunk 1 above)
Comment (kitallis) [reply src/honeysql_postgres/format.cljc:119]: @scimetfoo I agree for complex threads, but I felt like this was a simple enough case to fit in well. Although my usage was clearly broken, so what do you think of threading all the other expressions except for the conditional `as->`?

Comment (scimetfoo) [reply src/honeysql_postgres/format.cljc:119]: @kitallis what are your thoughts about this?

```
(defmethod format-clause :filter [[_ expr] sql-map]
  (let [format (fn [expr]
                 (let [[expression clause alias] (mapv sqlf/to-sql expr)]
                   (->> alias
                        (str " AS ")
                        (when alias)
                        (str expression " FILTER " clause))))]
    (->> expr
         (map format)
         sqlf/comma-join
         (str (when (seq (:select sql-map)) ", ")))))
```

Comment (kitallis) [reply src/honeysql_postgres/format.cljc:119]: @scimetfoo Yup looks good to me 👍🏼

**Review by kitallis — DISMISSED** [review]

**Review by kitallis — APPROVED** [review]
